env/io_posix.cc

```diff
@@ -95,8 +95,9 @@ size_t GetLogicalBufferSize(int __attribute__((__unused__)) fd) {
   if (parent_begin == std::string::npos) {
     return kDefaultPageSize;
   }
-  if (device_dir.substr(parent_begin + 1, parent_end - parent_begin - 1) !=
-      "block") {
+  std::string parent = device_dir.substr(parent_begin + 1, parent_end - parent_begin - 1);
+  std::string child = device_dir.substr(parent_end + 1, device_dir.length() - 1);
+  if (parent != "block" && (child.compare(0, 4, "nvme") || child.find('p') != std::string::npos)) {
```

Comment:
If `parent != "block"` AND `child.compare(0, 4, "nvme") == 0` (starts with "nvme"), then we have to evaluate `child.find('p')`. If child has 'p', then we are going to execute inside the `if`. In this case, does it mean it's not an nvme even if child starts with "nvme"?

Reply:
just to remove the last child, `"nvme0n1p1"` out of `"nvme/nvme0/nvme0n1/nvme0n1p1"` if nvme drive is partitioned.

Reply:
Got it, thanks

Reply:
Good question. This below would explain partitioned nvme drive naming convention as I just crawled into linux kernel source code.
```
linux-stable/block/partition-generic.c
struct hd_struct *add_partition(struct gendisk *disk, int partno,
                                sector_t start, sector_t len, int flags,
                                struct partition_meta_info *info)
{
...
        dname = dev_name(ddev);
        if (isdigit(dname[strlen(dname) - 1]))
                dev_set_name(pdev, "%sp%d", dname, partno);
        else
                dev_set_name(pdev, "%s%d", dname, partno);
...
}
```